components/sup/src/event.rs

```diff
@@ -237,47 +231,20 @@ fn publish(mut event: impl EventMessage) {
                                                  Some(std::time::SystemTime::now().into()),
                                              ..EVENT_CORE.get::<EventCore>().to_event_metadata() });
 
-        e.send(event.to_bytes());
+        e.lock().send(event.to_bytes());
     }
 }
 
 /// A lightweight handle for the event stream. All events get to the
 /// event stream through this.
-struct EventStream(UnboundedSender<Vec<u8>>);
+struct EventStream(Sender<Vec<u8>>);
 
 impl EventStream {
     /// Queues an event to be sent out.
-    fn send(&self, event: Vec<u8>) {
+    fn send(&mut self, event: Vec<u8>) {
         trace!("About to queue an event: {:?}", event);
-        if let Err(e) = self.0.unbounded_send(event) {
-            error!("Failed to queue event: {:?}", e);
+        if let Err(e) = self.0.try_send(event) {
```

Comment:
The current error we message we give when `try_send` fails due to a full channel is `Failed to queue event: send failed because channel is full`. Is there something more you would like to see?

Reply:
Nope, that's good!

Reply:
I think a comment in the code about dropping messages would still be useful as a documentation of intent.